### PR TITLE
Modification for LB, FW resources for unique name, by having provider…

### PR DESCRIPTION
**Pull request by ashutosh-mishra**
… PTG name in resources names.

### CODE REVIEW

**Comment by ashutosh-mishra** [comment]
@chandra-k

File: gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/heat_node_driver.py [hunk 1]
```diff
@@ -387,6 +387,15 @@ def _update_template_with_firewall_rules(self, context, provider_ptg,
                             stack_template[resources_key],
                             is_template_aws_version,
                             resource_name)
+        fw_resource_name = 'OS::Neutron::Firewall'
+        fw_key = self._get_heat_resource_key(
+                            stack_template[resources_key],
+                            is_template_aws_version,
+                            fw_resource_name)
+        # Include provider name in firewall, firewall policy.
```
Comment (rukansari) [inline gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/heat_node_driver.py]: I don't see the update for firewall policy here as stated in the comment

File: gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/heat_node_driver.py [hunk 2]
```diff
@@ -387,6 +387,15 @@ def _update_template_with_firewall_rules(self, context, provider_ptg,
...
+        ptg_name = '-' + provider_ptg['name']
+        stack_template[resources_key][fw_policy_key][properties_key]['name'] += ptg_name
+        stack_template[resources_key][fw_key][properties_key]['name'] += ptg_name
```
Comment (rukansari) [inline gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/heat_node_driver.py]: please make this modular by introducing a method to update name  like the lb case.
Then the same method can be re-used by OC service node driver

File: gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py [hunk 3]
```diff
@@ -816,7 +816,22 @@ def _update_template_with_firewall_rules(self, context, provider_ptg,
...
+                            stack_template[resources_key],
+                            is_template_aws_version,
+                            fw_resource_name)
+            # Include provider name in firewall, firewall policy.
+            ptg_name = '-' + provider_ptg['name']
+            stack_template[resources_key][fw_policy_key][properties_key]['name'] += ptg_name
+            stack_template[resources_key][fw_key][properties_key]['name'] += ptg_name
+
+            stack_template[resources_key][fw_policy_key][properties_key][
                 'firewall_rules'] = fw_rule_list
```
Comment (rukansari) [inline gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py]: replicated code from heat node driver. Re-use it from heat node driver by making it a common method

**Comment by rukansari** [comment]
Suggest review by Magesh/Vikash prior to merging this

**Comment by ashutosh-mishra** [comment]
Closing the PR, As we don't need to modify heat_node_driver.py, we only need modifications in oneconvergence_service_node_driver.py file.
Reopened another PR https://github.com/oneconvergence/group-based-policy/pull/15 for changes, Please review it.

**Pull request closed without merging**